src/main/java/org/candlepin/resource/RhsmApiCompatResource.java

```diff
@@ -0,0 +1,109 @@
+package org.candlepin.resource;
+
+import org.candlepin.dto.api.server.v1.ConsumerEntitlementCountsDTO;
+import org.candlepin.dto.api.server.v1.ConsumerEntitlementCountsDTOAllOfEntitlementCounts;
+import org.candlepin.dto.api.server.v1.RhsmApiConsumerEntitlementCountsQueryDTO;
+import org.candlepin.model.ConsumerEntitlementCount;
+import org.candlepin.model.RhsmApiCompatCurator;
+import org.candlepin.resource.server.v1.RhsmapiApi;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Entry point for endpoints that are dedicated for RHSM API for compatibility reasons.
+ */
+public class RhsmApiCompatResource implements RhsmapiApi {
+    private final RhsmApiCompatCurator rhsmApiCompatCurator;
+
+    @Inject
+    public RhsmApiCompatResource(RhsmApiCompatCurator rhsmApiCompatCurator) {
+        this.rhsmApiCompatCurator = Objects.requireNonNull(rhsmApiCompatCurator);
+    }
+
+    // POST /rhsmapi/consumers/entitlement_counts
+
+    @Override
+    @Transactional
+    public Stream<ConsumerEntitlementCountsDTO> listConsumerEntitlementCounts(
+        @Valid @NotNull RhsmApiConsumerEntitlementCountsQueryDTO rhsmApiConsumerEntitlementCountsQueryDTO,
+        Integer page, Integer perPage) {
+
+        List<String> consumerIds = rhsmApiConsumerEntitlementCountsQueryDTO.getConsumerIds();
+        List<String> consumerUuids = rhsmApiConsumerEntitlementCountsQueryDTO.getConsumerUuids();
+
+        List<ConsumerEntitlementCount> consumerEntCounts = this.rhsmApiCompatCurator
+            .getConsumerEntitlementCounts(consumerIds, consumerUuids);
+
+        // The order of the ConsumerEntitlementCountsDTOs needs to be in the same order as consumerEntCounts
+        List<ConsumerEntitlementCountsDTO> consumerEntCountDTOs = new ArrayList<>();
+        ConsumerEntitlementCountsDTO current = null;
+        for (ConsumerEntitlementCount consumerEntCount : consumerEntCounts) {
+            // Base case for the first iteration
+            if (current == null) {
+                current = new ConsumerEntitlementCountsDTO()
+                    .consumerId(consumerEntCount.id())
+                    .consumerUuid(consumerEntCount.uuid());
+            }
+
+            // We are now listing entitlement counts for a different consumer, so we can start populating
+            // a new ConsumerEntitlementCountsDTO for this consumer
+            if (!current.getConsumerId().equals(consumerEntCount.id())) {
+                consumerEntCountDTOs.add(current);
+
+                current = new ConsumerEntitlementCountsDTO()
+                    .consumerId(consumerEntCount.id())
+                    .consumerUuid(consumerEntCount.uuid());
+            }
+
+            ConsumerEntitlementCountsDTOAllOfEntitlementCounts entitlementCountsItem =
```

Comment:
Is this actually what the dto generator came up with for a name here? If so, gross. We're better off defining it ourselves with something that's sane and won't force line breaking early.

`RhsmApiConsumerEntitlementCountDTO` has `RhsmApiEntitlementCountDTO`